Sources/EdgeNetworkHandlers/EdgeHitProcessor.swift

```diff
@@ -74,12 +77,8 @@ class EdgeHitProcessor: HitProcessing {
         let requestBuilder = RequestBuilder()
         requestBuilder.enableResponseStreaming(recordSeparator: EdgeConstants.Defaults.RECORD_SEPARATOR,
                                                lineFeed: EdgeConstants.Defaults.LINE_FEED)
-
-        if let ecid = identityState[EdgeConstants.SharedState.Identity.ECID] as? String {
-            requestBuilder.experienceCloudId = ecid
-        } else {
-            // This is not expected to happen. Continue without ECID
-            Log.warning(label: LOG_TAG, "processHit - An unexpected error has occurred, ECID is nil.")
+        if let identityMap = identityState[EdgeConstants.SharedState.Identity.IDENTITY_MAP] as? [String: Any] {
```

Comment:
Knowing the format of the IdentityMap is not optimal here. In a perfect situation, we could blindly attach the shared state, however, there is a bit of a limitation with `Encodable. We must know the keys we are going to encode before we encode them.

We should create a task for future work to do some fancy encoding so we can dynamically key our encoding container based on which ever XDM payloads are attached.

Reply:
Nevermind, I came up with a solution for not knowing the CodingKeys ahead of time. https://github.com/adobe/aepsdk-edge-ios/pull/165/commits/e7dfd3a3b52749bc74ed6b55730883cfb2361065#diff-1c5046ba14094114b426fa3ad5722b3901cb88b555ffdfdef0beb37683d51f1bR32